packages/manager/src/components/TagsPanel/TagsPanel.spec.js

```diff
@@ -2,48 +2,169 @@ const { navigateToStory, executeInAllStories } = require('../../../e2e/utils/sto
+      .withContext(`add tag button should not exist`)
+      .toBe(false);
+    createTagSelect.setValue(tagname);
+    createTagSelect.addValue('\uE007');
+    browser.waitUntil(() => {
+      return $$(tag).length == startTags + 1
+    }, constants.wait.normal);
+  }
+
+  const verifyTagName = (tagname) => {
+    expect($(`[data-qa-tag="${tagname}"]`).isExisting())
+      .withContext(`incorrect tag name`)
+      .toBe(true);
+    expect($(`[data-qa-tag="${tagname}"] span`).getText())
+      .withContext(`incorrect span text`)
+      .toBe(tagname);
+  }
+
+  const deleteTagName = (tagname) => {
+    const tags = $$(tag).length
+    $(`[data-qa-tag="${tagname}"] [data-qa-delete-tag]`).click();
+    browser.waitUntil(() => {
+      return $$(tag).length < tags
+    }, constants.wait.normal);
+    expect($(`[data-qa-tag="${tagname}"]`).isExisting())
+      .withContext(`${tagname} should not exist in the DOM`)
+      .toBe(false)
+  }
+
+  beforeEach(() => {
+    navigateToStory(component, childStories[0]);
+    $(addTag).waitForDisplayed(constants.wait.normal);
+  });
+
+  it('there should be a tag panel, and icons to add and delete tags', () => {
+    $(tag).waitForDisplayed(constants.wait.normal);
+    expect($$(tag).length)
+      .withContext(`should be 3 tags`)
+      .toBe(3);
+    expect($$(deleteTag).length)
+      .withContext(`should be 3 delete tag icons`)
+      .toBe(3);
+    expect($(addTag).isDisplayed())
+      .withContext(`plus symbol should be displayed`)
+      .toBe(true);
+    expect($(addTagParagraph).isDisplayed())
+      .withContext(`'Add New Tag' p element should be displayed`)
+      .toBe(true);
+  });
+
+  it('can open menu with plus button', () => {
+    executeInAllStories(component,childStories, () => {
+      $(addTagParagraph).click();
+      expect($(tagsSelect).isDisplayed())
+        .withContext(`tag select should be displayed`)
+        .toBe(true);
+      expect($(addTagParagraph).isExisting())
+        .withContext(`add new tag text should not exist`)
+        .toBe(false);
+      expect($(addTag).isExisting())
+        .withContext(`add tag button should not exist`)
+        .toBe(false);
+    })
+  });
+
+ it('can open menu with Add New tag text', () => {
+    executeInAllStories(component,childStories, () => {
+      $(addTag).click();
+      expect($(tagsSelect).isDisplayed())
+        .withContext(`tag select should be displayed`)
+        .toBe(true);
+      expect($(addTagParagraph).isExisting())
+        .withContext(`add new tag text should not exist`)
+        .toBe(false);
+      expect($(addTag).isExisting())
+        .withContext(`add tag button should not exist`)
+        .toBe(false);
+    })
+  });
+
+  describe('Adding Tags', () => {
+    it('can add tags', () => {
+      executeInAllStories(component,childStories, () => {
+        const testTag = "TEST_TAG";
+        addNewTag(testTag)
+        expect($$(tag).find(tagName => tagName.getText() === testTag))
```

Comment:
maybe instead of a `not.toBeNull` assertion, we can update it to be 

```javascript
const matchingTag = $$(tag).find(tagName => tagName.getText() === testTag)

expect(matchingTag.length).toBe(1)
expect(matchingTag.getText()).toBe(testTag)
```